ProcessLib/VectorMatrixAssembler.cpp

```diff
@@ -160,4 +197,34 @@ void VectorMatrixAssembler::assembleWithJacobian(
     }
 }
 
+void VectorMatrixAssembler::localAssembleWithJacobianAndCoupling(
+    const double t, std::vector<GlobalIndexType> const& base_indices,
+    std::vector<GlobalIndexType> const& full_indices,
+    std::vector<double> const& local_xdot,
+    LocalAssemblerInterface& local_assembler, const double dxdot_dx,
+    const double dx_dx, CoupledSolutionsForStaggeredScheme const* cpl_xs)
+{
+    std::vector<std::reference_wrapper<const std::vector<GlobalIndexType>>>
+        indices_of_all_coupled_processes;
+    indices_of_all_coupled_processes.reserve(cpl_xs->coupled_xs.size());
+    for (std::size_t i = 0; i < cpl_xs->coupled_xs.size() - 1; i++)
+    {
+        indices_of_all_coupled_processes.emplace_back(std::ref(base_indices));
+    }
+    indices_of_all_coupled_processes.emplace_back(std::ref(full_indices));
```

Comment:
Just curious, does using of ("slightly" awkward) reference_wrapper pays of? I'd think the indices vector is rather short, isn't it?

Reply:
Changed. See my previous reply.